### PR TITLE
[command] reverify subcommand, reverify storage block data

### PR DESCRIPTION
**Pull request by 0xcb9ff9**
# Description

* Add `reverify`  command, verify storage block data from select start height, helper fix account nonce error (`trie` database breakage)
* [replace `network.Server` to interface](https://github.com/dogechain-lab/dogechain/pull/274/commits/5e9c46dbc911bc27eb23d2e215af1f236e4b415b)
* `kvdb` add `Iterator` API

# Changes include

- [x] New feature (non-breaking change that adds functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

use 
```
$ dogechain reverify --chain genesis.json --data-dir [Data Dir] --start-height 50
```

# Additional comments

```
Reverify block data

Usage:
   reverify [flags]

Flags:
      --chain string           the genesis file path (default "./genesis.json")
      --data-dir string        the data directory used for storing Dogechain-Lab Dogechain client data
  -h, --help                   help for verify
      --pprof                  enable the pprof server
      --pprof-address string   the address and port for the pprof service (default "127.0.0.1:6060")
      --start-height string    start verify block height (default "1")

Global Flags:
      --json   get all outputs in json format (default false)
```

### CODE REVIEW

**Review by DarianShawn — DISMISSED** [review]
LGTM.
The only flaw is the name of this command, it seems to be better to use `restate` or `reverify`.

**Comment by 0xcb9ff9** [comment]
@DarianShawn ok, rename to `reverify`

File: helper/kvdb/kvdb.go [hunk 1]
```diff
@@ -5,6 +5,54 @@ type KVBatch interface {
 	Write() error
 }
 
+type KVIteratioRange struct {
```
Comment (DarianShawn) [inline helper/kvdb/kvdb.go]: typo

**Review by DarianShawn — DISMISSED** [review]
LGTM
The abstraction of network and server is good.
Only got a typo to fix.

File: helper/kvdb/kvdb.go (hunk 1 above)
Comment (0xcb9ff9) [reply helper/kvdb/kvdb.go]: @DarianShawn  done, add more test [a365215](https://github.com/dogechain-lab/dogechain/pull/274/commits/a365215002a62b23f6cea9461b2e49546d9836d6)

**Review by DarianShawn — APPROVED** [review]

**Pull request merged**